frontend/src/app/api/http-internal.service.ts

```diff
@@ -0,0 +1,57 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { environment } from 'src/environments/environment';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class HttpInternalService {
+    private apiUrl: string = environment.apiUrl;
+
+    constructor(private http: HttpClient) {}
+
+    public getRequest<T>(url: string, httpParams?: any): Observable<T> {
```

Comment:
Why do you need request word in method names?

Reply:
Fixed